app/services/direct-link.js

```diff
@@ -0,0 +1,87 @@
+import { getOwner } from '@ember/application';
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+
+// TODO: find a better way to store / configure the handlers map
+const REDIRECT_HANDLER = {
+  'http://www.w3.org/ns/person#Person': 'person',
+  'http://data.lblod.info/vocabularies/erediensten/BestuurVanDeEredienst':
+    'administrative-unit',
+  'http://data.lblod.info/vocabularies/erediensten/CentraalBestuurVanDeEredienst':
+    'administrative-unit',
+};
+
+export default class DirectLinkService extends Service {
+  @service uriInfo;
+
+  REDIRECT_HANDLER = REDIRECT_HANDLER;
+
+  async redirect(resourceUri) {
+    let uriInfo = await this.uriInfo.getDirectLinks(resourceUri);
+    let rdfType = getRdfType(uriInfo);
+    let uuid = getUuid(uriInfo);
+
+    if (rdfType && uuid) {
+      this._handleRedirect(rdfType, uuid, resourceUri);
+    } else {
+      throw new ResourceNotFoundError(
+        `No data found for ${resourceUri}`,
+        resourceUri
+      );
+    }
+  }
+
+  async _handleRedirect(type, uuid, resourceUri) {
+    let handlerName = this.REDIRECT_HANDLER[type];
+    let redirectHandler = getOwner(this).lookup(
+      `redirect-handler:${handlerName}`
+    );
+
+    if (redirectHandler) {
+      await redirectHandler.redirect(uuid, resourceUri);
+    } else {
+      throw new RedirectHandlerNotFoundError(
+        `No redirect handler registered for "${type}"`,
+        {
+          type,
+          uuid,
+          resourceUri,
+        }
+      );
+    }
+  }
+}
+
+function getRdfType(uriInfo) {
+  const RDF_TYPE_URL = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#type';
+  let triple = uriInfo.triples.find(
```

Comment:
This assumes there will only be 1 type which isn't always the case. I guess we should loop through all the results until a redirect-handler is found?

Reply:
Added support for multiple rdf types. Now the first one that is mapped to a RedirectHandler class will be used.